### PR TITLE
Add http async client discovery

### PR DESCRIPTION
**Pull request by joelwurtz**

### CODE REVIEW

File: src/HttpAsyncClientDiscovery.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Finds an HTTP Asynchronous Client
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class HttpAsyncClientDiscovery extends ClassDiscovery
+{
+    /**
+     * @var HttpAsyncClient
```
Comment (sagikazarmark) [inline src/HttpAsyncClientDiscovery.php]: Shouldn't this be imported?

File: src/HttpAsyncClientDiscovery.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     */
+    protected static $cache;
+
+    /**
+     * @var array
+     */
+    protected static $classes = [
+        'guzzle6' => [
+            'class'     => 'Http\Adapter\Guzzle6HttpAdapter',
+            'condition' => 'Http\Adapter\Guzzle6HttpAdapter'
```
Comment (sagikazarmark) [inline src/HttpAsyncClientDiscovery.php]: Missing comma for the last element (if this is copy&paste, then it is probably missing for the original discovery as well.

File: src/HttpAsyncClientDiscovery.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        ],
+    ];
+
+    /**
+     * Finds an HTTP Async Adapter
```
Comment (sagikazarmark) [inline src/HttpAsyncClientDiscovery.php]: Client. Copy&Paste?

File: src/HttpAsyncClientDiscovery.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Http\Discovery;
+
+/**
+ * Finds an HTTP Asynchronous Client
```
Comment (sagikazarmark) [inline src/HttpAsyncClientDiscovery.php]: Async term should be consistent

File: src/HttpAsyncClientDiscovery.php (hunk 3 above)
Comment (joelwurtz) [inline src/HttpAsyncClientDiscovery.php]: yes :)

File: spec/HttpAsyncClientDiscoverySpec.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        $this->shouldHaveType('Http\Discovery\HttpAsyncClientDiscovery');
+    }
+
+    function it_is_a_class_discovery()
+    {
+        $this->shouldHaveType('Http\Discovery\ClassDiscovery');
+    }
+
+    function it_finds_an_http_adapter()
```
Comment (sagikazarmark) [inline spec/HttpAsyncClientDiscoverySpec.php]: client

File: src/HttpAsyncClientDiscovery.php (hunk 1 above)
Comment (joelwurtz) [inline src/HttpAsyncClientDiscovery.php]: should be good

Comment (sagikazarmark) [inline src/HttpAsyncClientDiscovery.php]: I mean the class

Comment (joelwurtz) [inline src/HttpAsyncClientDiscovery.php]: with a use ?

Comment (sagikazarmark) [inline src/HttpAsyncClientDiscovery.php]: Yes, otherwise IDEs won't find it.

Comment (joelwurtz) [inline src/HttpAsyncClientDiscovery.php]: Like i said it's done, github just don't show it here :) as it's from a old commit

File: src/HttpAsyncClientDiscovery.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Http\Discovery;
+
+use Http\Client\HttpAsyncClient;
```
Comment (joelwurtz) [inline src/HttpAsyncClientDiscovery.php]: poke @sagikazarmark

File: src/HttpAsyncClientDiscovery.php (hunk 1 above)
Comment (sagikazarmark) [inline src/HttpAsyncClientDiscovery.php]: Ah, great.

File: src/HttpAsyncClientDiscovery.php (hunk 6 above)
Comment (sagikazarmark) [inline src/HttpAsyncClientDiscovery.php]: Couldn't see it.

**Pull request merged**